src/storage/kv/oracle.rs

```diff
@@ -400,8 +408,8 @@ impl Mark {
         })
     }
 
-    fn take(&self) -> Sender<()> {
-        self.ch.lock().take().unwrap()
+    async fn take(&self) -> Sender<()> {
```

Comment:
We should allow synchrnonous alternatives alongside the async default ones. `try_take` and `take_blocking` for example. But that can be in a separate PR obviously, or even further down the line. I just know they will be useful to avoid sync/async interleaving.

Reply:
I agree with you, I am just keeping this as an optional PR, will work on what you said in this